src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseResource.java

```diff
@@ -700,4 +710,40 @@ private void validateStaticCodeAnalysisSettings(ProgrammingExercise programmingE
         return updateProgrammingExercise(programmingExercise, null);
     }
 
+    /**
+     * PUT /programming-exercises/:exerciseId/tasks : Parse tasks and corresponding test cases from problem statement on an existing ProgrammingExercise.
+     * @param exerciseId of the exercise
+     * @return the {@link ResponseEntity} with status {@code 200} and with body the tasks with test cases,
+     * or with status {@code 400 (Bad Request) if the exerciseId is not valid}.
+     */
+    @PutMapping(EXTRACT_TASKS)
+    @PreAuthorize("hasRole('EDITOR')")
+    @FeatureToggle(Feature.ProgrammingExercises)
+    public ResponseEntity<Set<ProgrammingExerciseTask>> updateTasksFromProblemStatement(@PathVariable Long exerciseId) {
+        log.debug("REST request to create ProgrammingExerciseTasks from problem statement : {}", exerciseId);
+        ProgrammingExercise exercise = programmingExerciseRepository.findByIdElseThrow(exerciseId);
+        authCheckService.checkHasAtLeastRoleForExerciseElseThrow(Role.EDITOR, exercise, null);
+
+        programmingExerciseTaskService.updateTasksFromProblemStatement(exercise);
+        Set<ProgrammingExerciseTask> tasks = programmingExerciseTaskRepository.findByExerciseIdWithTestCases(exerciseId);
+        return ResponseEntity.ok(tasks);
+    }
+
+    /**
+     * DELETE /programming-exercises/:exerciseId/tasks : Delete all tasks and solution entries for an existing ProgrammingExercise
+     * @param exerciseId of the exercise
+     * @return the {@link ResponseEntity} with status {@code 204},
+     * or with status {@code 400 (Bad Request) if the exerciseId is not valid}.
+     */
+    @DeleteMapping(TASKS)
```

Comment:
I agree that is not restful. But since this endpoint is only there for testing purposes and will be deleted at a later stage of our thesis, I would leave it like it is.
This is also discussed and agreed upon with Stephan.

Reply:
I see, would you mind adding a comment to it? Marking it for deletion?

Reply:
Added.